cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -366,6 +366,15 @@ angular.module(PKG.name + '.commons')
 
     function initSplitterNode(node) {
       if (!node.outputSchema || !Array.isArray(node.outputSchema) || (Array.isArray(node.outputSchema) && node.outputSchema[0].name === GLOBALS.defaultSchemaName)) {
+        let splitterPorts = splitterNodesPorts[node.name];
+        if (!_.isEmpty(splitterPorts)) {
```

Comment:
What happens when it's just the number of ports changed instead of clearing all?

Reply:
If we have multiple ports then the code wouldn't enter this big `if` condition, since the port names would always be different from `GLOBALS.defaultSchemaName`. This part of the code is only for when the splitter node doesn't have multiple output schemas yet, or when we clear them all.